modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -467,13 +467,67 @@ impl LightningClientModule {
         }
     }
 
+    pub async fn add_gateway(&self, gateway_key: PublicKey, gateway: SafeUrl) -> bool {
+        let mut dbtx = self.client_ctx.module_db().begin_transaction().await;
+        let new_entry = dbtx
+            .insert_entry(&GatewayKey(gateway_key), &gateway)
+            .await
+            .is_none();
+        dbtx.commit_tx().await;
+        new_entry
+    }
+
+    pub async fn delete_gateway(&self, gateway: SafeUrl) -> bool {
+        let mut dbtx = self.client_ctx.module_db().begin_transaction().await;
+        let gateways = dbtx
+            .find_by_prefix(&GatewayPrefix)
+            .await
+            .collect::<Vec<_>>()
+            .await;
+        let mut deleted = false;
+        for (key, url) in gateways {
+            if url == gateway {
+                deleted = dbtx.remove_entry(&key).await.is_some();
+            }
+        }
+
+        dbtx.commit_tx().await;
+        deleted
+    }
+
+    pub async fn list_all_gateways(&self) -> Result<Vec<SafeUrl>, SelectGatewayError> {
+        let mut gateways = self
+            .module_api
+            .get_registered_gateways()
+            .await
+            .map_err(|e| SelectGatewayError::FederationError(e.to_string()))?;
+        let mut dbtx = self.client_ctx.module_db().begin_transaction_nc().await;
+        let mut gateways_in_db = dbtx
+            .find_by_prefix(&GatewayPrefix)
+            .await
+            .filter_map(|(_, url)| async {
+                // Remove duplicates by filtering the gateways that are registered and in the db
+                if gateways.contains(&url) {
+                    None
+                } else {
+                    Some(url)
+                }
+            })
+            .collect::<Vec<_>>()
+            .await;
+
+        gateways.append(&mut gateways_in_db);
+
+        Ok(gateways)
+    }
+
     async fn select_gateway(
         &self,
         invoice: Option<Bolt11Invoice>,
     ) -> Result<(SafeUrl, RoutingInfo), SelectGatewayError> {
         let gateways = self
             .module_api
-            .gateways()
+            .get_registered_gateways()
```

Comment:
Do we even use the DB gateways right now? Seems to me that we'd need to add them here somehow.

Reply:
Integrators need to call `update_gateway_map` for the db to be updated currently. The CLI does not do this, so the CLI doesn't use the DB at all. I think this PR helps the CLI too